spec/index.bs

```diff
@@ -247,116 +247,120 @@ dictionary GPUObjectDescriptorBase {
 
-An <dfn dfn>adapter</dfn> represents an implementation of WebGPU on the system.
-Each adapter identifies both an instance of a hardware accelerator (e.g. GPU or CPU) and
-an instance of a browser's implementation of WebGPU on top of that accelerator.
-
-If an [=adapter=] becomes unavailable, it becomes [=invalid=].
-Once invalid, it never becomes valid again.
-Any [=devices=] on the adapter, and [=internal objects=] owned by those devices,
-also become invalid.
-
-Note:
-An [=adapter=] may be a physical display adapter (GPU), but it could also be
-a software renderer.
-A returned [=adapter=] could refer to different physical adapters, or to
-different browser codepaths or system drivers on the same physical adapters.
-Applications can hold onto multiple [=adapters=] at once (via {{GPUAdapter}})
-(even if some are [=invalid=]),
-and two of these could refer to different instances of the same physical
-configuration (e.g. if the GPU was reset or disconnected & reconnected).
-
-### GPUAdapter ### {#gpu-adapter}
+<div algorithm=requestAdapter>
+    **Arguments:**
+      - optional {{GPURequestAdapterOptions}} |options| = {}
 
-A <dfn interface>GPUAdapter</dfn> refers to an [=adapter=],
-and exposes its capabilities (extensions and limits).
-
-<script type=idl>
-interface GPUAdapter {
-    readonly attribute DOMString name;
-    readonly attribute sequence<GPUExtensionName> extensions;
-    //readonly attribute GPULimits limits; Don't expose higher limits for now.
+    **Returns:** |promise|, of type Promise<{{GPUAdapter}}>.
 
-    // May reject with DOMException  // TODO: DOMException("OperationError")?
-    Promise<GPUDevice> requestDevice(optional GPUDeviceDescriptor descriptor = {});
-};
-</script>
+    Requests an [=adapter=] from the user agent.
+    The user agent chooses whether to return an adapter, and, if so,
+    chooses according to the provided |options|.
 
-{{GPUAdapter}} has the following attributes:
+    Returns a new Promise, |promise|.
 
-<dl dfn-type=attribute dfn-for=GPUAdapter>
-    : <dfn>name</dfn>
-    ::
-        A human-readable name identifying the adapter.
-        The contents are implementation-defined.
+      - If the user agent chooses to return an adapter,
+        |promise| resolves with a new {{GPUAdapter}} encapsulating
+        a new [=adapter=].
+      - Otherwise, |promise| rejects with an {{OperationError}}.
 
-    : <dfn>extensions</dfn>
-    ::
-        A sequence containing the {{GPUExtensionName}}s of the extensions supported by the adapter.
-</dl>
-
-{{GPUAdapter}} also has the following internal slots:
-
-<dl dfn-type=attribute dfn-for=GPUAdapter>
-    : <dfn>\[[adapter]]</dfn>, of type [=adapter=], readonly
-    ::
-        An internal slot holding the [=adapter=] to which this {{GPUAdapter}}
-        refers.
-</dl>
-
-### Getting an Adapter ### {#adapter-creation}
-
-To get a {{GPUAdapter}}, use {{GPU/requestAdapter()}}.
+    Issue: Figure out how to integrate the options.
```

Comment:
Can the issue describe in more details what needs to be done?

Reply:
I didn't because I hadn't figured out what needed to be done. I'm going to leave this in and open a new PR that actually does the integration (hopefully).